src/python/pants/engine/internals/options_parsing.py

```diff
@@ -58,6 +60,16 @@ def log_level(global_options: GlobalOptions) -> LogLevel:
     return global_options.level
 
 
+@rule
+def extract_process_cleanup_option(keep_sandboxes: KeepSandboxes) -> ProcessCleanupOption:
+    warn_or_error(
```

Comment:
I asked about this elsewhere, but is this a new thing? Deprecating rule APIs? It's long overdue, but my initial grok of this when shaving this yak to un-break a user support issue was that it was a mis-placed deprecation for the corresponding option. If this is now going to be something to expect to see in rules I'll adjust my mental model.

Reply:
I'd say that the current situation is "best effort basis". It was easy to do in this case, so I did... but yea, it's very not required.